go/api/cluster_client.go

```diff
@@ -0,0 +1,18 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+package api
+
+// RedisClusterClient is a client used for connection to cluster Redis servers.
+type RedisClusterClient struct {
+	*baseClient
+}
+
+// NewRedisClusterClient creates a [RedisClusterClient] in cluster mode using the given [RedisClusterClientConfiguration].
+func NewRedisClusterClient(config *RedisClusterClientConfiguration) (*RedisClusterClient, error) {
```

Comment:
Should be named `createClient` or `createClusterClient` to align with other clients